analyzers/src/SonarAnalyzer.VisualBasic/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -152,5 +164,70 @@ private static bool ShouldHandle(SyntaxNode method, SyntaxNode current)
+                    ? precedingConcat.Right
+                    : syntaxNode.Left;
+                return GetStringConstant(left) is { } leftString
+                    && GetStringConstant(syntaxNode.Right) is { } rightString
+                    ? leftString + rightString
+                    : null;
+
+                string GetStringConstant(SyntaxNode node) =>
+                    node.GetStringValue() ?? semanticModel.GetConstantValue(node).Value as string;
+            }
+
+            protected override string GetVariableName(BinaryExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(BinaryExpressionSyntax syntaxNode, SemanticModel semanticModel) => true;
+        }
+
+        private class InterpolatedStringBannedWordsFinder : CredentialWordsFinderBase<InterpolatedStringExpressionSyntax>
+        {
+            public InterpolatedStringBannedWordsFinder(DoNotHardcodeCredentialsBase<SyntaxKind> analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(InterpolatedStringExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.Contents.JoinStr(null, x => x switch
+                {
+                    InterpolationSyntax interpolation => semanticModel.GetConstantValue(interpolation.Expression).Value as string,
+                    InterpolatedStringTextSyntax text => text.TextToken.ToString(),
+                    _ => null
+                } ?? CredentialSeparator.ToString()); // Unknown elements resolved to separator to terminate the keyword-value sequence
+
+            protected override string GetVariableName(InterpolatedStringExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(InterpolatedStringExpressionSyntax syntaxNode, SemanticModel semanticModel) => true;
+        }
+
+        private class InvocationBannedWordsFinder : CredentialWordsFinderBase<InvocationExpressionSyntax>
+        {
+            public InvocationBannedWordsFinder(DoNotHardcodeCredentials analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(InvocationExpressionSyntax syntaxNode, SemanticModel semanticModel)
+            {
+                var allArgs = syntaxNode.ArgumentList.Arguments.Select(x => semanticModel.GetConstantValue(x.GetExpression()).Value as string ?? CredentialSeparator.ToString());
+                try
+                {
+                    return string.Format(allArgs.First(), allArgs.Skip(1).ToArray());
+                }
+                catch (FormatException)
+                {
+                    return null;
+                }
+            }
+
+            protected override string GetVariableName(InvocationExpressionSyntax syntaxNode) => null;
+
+            protected override bool ShouldHandle(InvocationExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.IsMethodInvocation(KnownType.System_String, "Format", semanticModel)
```

Comment:
If we opt to create the extension, this can be replaced too.

Reply:
_Note:_ Second extension would needed to be created for VB, as they don't share argument types.